Java/libraries/recognizers-text-number-with-unit/src/main/java/com/microsoft/recognizers/text/numberwithunit/parsers/BaseCurrencyParser.java

```diff
@@ -120,37 +125,22 @@ private ParseResult mergeCompoundUnit(ExtractResult compoundResult) {
                 } else {
                     // If the fraction unit doesn't match the main unit, finish process this group.
                     if (result != null) {
-                        if (mainUnitIsoCode == null || mainUnitIsoCode.isEmpty() ||
-                            mainUnitIsoCode.startsWith(Constants.FAKE_ISO_CODE_PREFIX)) {
-                            result.setValue(new UnitValue(String.valueOf(numberValue), mainUnitValue));
-                        } else {
-                            result.setValue(new CurrencyUnitValue(String.valueOf(numberValue), mainUnitValue, mainUnitIsoCode));
-                        }
-
-                        results.add(result);
-                        result = null;
+                        addToResults(result,mainUnitIsoCode,numberValue,mainUnitValue,results);
                     }
+                    result = null;
```

Comment:
This null assignment and the numberValue resest mean the same thing, right? Move this to the same location as in .cs
Nit: Also, add a blankline after the bracket close and before these assignments.